File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        List<Integer> list=new ArrayList<>();
        
        for(int n:people){
            list.add(n);
        }
        list.sort((a,b)->a-b);
        int left=0, right=people.length-1;
        int answer = 0;
        while(left<=right){
            if(list.get(left)+list.get(right)<=limit){
                left++;
            }
            right--;
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a,b)=>b-a);
    let left=0, right=people.length-1;
    let cnt=0;
    while(left<right){
        if(people[left]+people[right]<=limit) right--;
        left++;
        cnt++;
    }
    if(left===right) cnt++;
    return cnt;
}